docs/Explore Algorithms/Deep Learning/Quickstart - Chat Completion with Azure AI Foundry Model.ipynb

```diff
@@ -0,0 +1 @@
struct\"\n","api_key = find_secret(\n","    secret_name=\"synapseml-ai-foundry-resource-key\", keyvault=\"mmlspark-build-keys\"\n",")  # please replace this line with your key as a string\n","\n","assert api_key is not None and service_name is not None"],"outputs":[],"execution_count":null,"metadata":{"microsoft":{"language":"python","language_group":"synapse_pyspark"}},"id":"d65e5ef6-dc64-44e4-ae59-10ed8d8c1811"},{"cell_type":"markdown","source":["## Chat Completion\n","\n","Models such as Phi-4 and llama are capable of understanding chats instead of single prompts. The `AIFoundryChatCompletion` transformer exposes this functionality at scale."],"metadata":{"nteract":{"transient":{"deleting":false}},"microsoft":{"language":"python","language_group":"synapse_pyspark"}},"id":"886323f0-22c9-49c9-9c26-ca164c273c04"},{"cell_type":"code","source":["from synapse.ml.services.aifoundry import AIFoundryChatCompletion\n","from pyspark.sql import Row\n","from pyspark.sql.types import *\n","\n","\n","def make_message(role, content):\n","    return Row(role=role, content=content, name=role)\n","\n","\n","chat_df = spark.createDataFrame(\n","    [\n","        (\n","            [\n","                make_message(\n","                    \"system\", \"You are an AI chatbot with red as your favorite color\"\n","                ),\n","                make_message(\"user\", \"Whats your favorite color\"),\n","            ],\n","        ),\n","        (\n","            [\n","                make_message(\"system\", \"You are very excited\"),\n","                make_message(\"user\", \"How are you today\"),\n","            ],\n","        ),\n","    ]\n",").toDF(\"messages\")\n","\n","\n","chat_completion = (\n","    AIFoundryChatCompletion()\n","    .setSubscriptionKey(api_key)\n","    .setCustomServiceName(service_name)\n","    .setModel(model)\n","    .setApiVersion(\"2024-05-01-preview\")\n","    .setMessagesCol(\"messages\")\n","    .setErrorCol(\"error\")\n","    .setOutputCol(\"chat_completions\")\n",")\n","\n","display(\n","    chat_completion.transform(chat_df).select(\n","        \"messages\", \"chat_completions.choices.message.content\"\n","    )\n",")"],"outputs":[],"execution_count":null,"metadata":{"microsoft":{"language":"python","language_group":"synapse_pyspark"},"collapsed":false},"id":"d4491222-78c6-4136-8711-8f837915b2a9"}],"metadata":{"language_info":{"name":"python"},"kernel_info":{"name":"synapse_pyspark"},"kernelspec":{"display_name":"synapse_pyspark","language":null,"name":"synapse_pyspark"},"a365ComputeOptions":null,"sessionKeepAliveTimeout":0,"microsoft":{"language":"python","language_group":"synapse_pyspark","ms_spell_check":{"ms_spell_check_language":"en"}},"nteract":{"version":"nteract-front-end@1.0.0"},"spark_compute":{"compute_id":"/trident/default","session_options":{"conf":{"spark.synapse.nbs.session.timeout":"1200000"}}},"synapse_widget":{"version":"0.1","state":{}},"dependencies":{"lakehouse":null}},"nbformat":4,"nbformat_minor":5}
```

Comment:
if we can format this so its multiline that will make it a bit easier to comment on